boltstub/packstream.py

```diff
@@ -562,6 +569,20 @@ def _to_jolt_v3_type(self):
             dtype_marker, data = self.fields
             dtype = V3_VECTOR_DTYPE[dtype_marker]
             return jolt_v3_types.JoltVector(dtype, data)
+        if self.tag == StructTagV3.unsupported:
+            if len(self.fields) == 4:
+                (
+                    name, protocol_minimum_major,
+                    protocol_minimum_minor, extra
+                ) = self.fields
+                return jolt_v3_types.JoltUnsupportedType(
+                    name, protocol_minimum_major,
+                    protocol_minimum_minor, extra["message"]
+                )
+            name, protocol_minimum_major, protocol_minimum_minor = self.fields
+            return jolt_v3_types.JoltUnsupportedType(
+                name, protocol_minimum_major, protocol_minimum_minor
+            )
```

Comment:
According to the ADR, the struct always has 4 fields, but `extra["message"]` might result in a `KeyError` :upside_down_face: